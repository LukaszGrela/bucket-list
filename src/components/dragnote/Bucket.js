import React from 'react';
import PropTypes from 'prop-types';
import AddBucketButton from './AddBucketButton';
import Note from './Note';
import { DropTarget } from 'react-dnd';
import { ItemTypes } from './ItemTypes';
import './styles/Bucket.scss';



class Bucket extends React.Component {

    addNote = () => {
        const { addNoteHandler, id } = this.props;
        addNoteHandler && addNoteHandler(id);
    }
    render = () => {
        const { id, connectDropTarget,
            name, className,
            addNoteHandler, notes = { list: [] } } = this.props;
        const list = notes.list.filter(note => note.bucketId === id);
        return connectDropTarget(

            <div className={'bucket' + (className ? ' ' + className : '')}>
                <div className='bucket-name'>{name}</div>
                <div className='notes-container'>
                    {
                        notes.loading ? "Notes loading..." :
                            list.map((note, index) => <Note key={note.id} {...note} />)

                    }
                    <div className='add-note'>
                        <AddBucketButton handleClick={this.addNote} />
                    </div>
                </div>
                <div className='bucket-toolbar'>Add Note</div>
            </div>
        )
    };
}
Bucket.propTypes = {
    addNoteHandler: PropTypes.func.isRequired,
    onDrop: PropTypes.func.isRequired,
    isOver: PropTypes.bool.isRequired,
    canDrop: PropTypes.bool.isRequired,
    connectDropTarget: PropTypes.func.isRequired,
};

const dropZoneTarget = {
    canDrop: (props, monitor) => {
        return true;
    },
    drop: (props, monitor) => {
        props.onDrop(props.id, monitor.getItem())
    }
};
const collect = (connect, monitor) => ({
    connectDropTarget: connect.dropTarget(),
    isOver: monitor.isOver(),
    canDrop: monitor.canDrop()
});

export default DropTarget(ItemTypes.NOTE,
    dropZoneTarget,
    collect)(Bucket);