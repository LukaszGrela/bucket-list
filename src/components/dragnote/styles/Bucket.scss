.bucket {
    width: 20rem;
    display: inline-block;
    min-height: 24rem;
    background-color: #ccc;
    border-radius: 0.6rem;
    margin: 1rem;
    padding: 0.6rem;
    overflow: hidden;
}