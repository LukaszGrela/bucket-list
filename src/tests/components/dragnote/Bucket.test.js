import React from 'react';
import { shallow } from 'enzyme';
import Bucket from '../../../components/dragnote/Bucket';

const identity = el => el;
const OriginalBucket = Bucket.DecoratedComponent;

let wrapper, addNoteHandler, onDrop, isOver, canDrop;

beforeEach(() => {
    addNoteHandler = jest.fn();
    onDrop = jest.fn();
    isOver = false;
    canDrop = false;

    wrapper = shallow(<OriginalBucket
        addNoteHandler={addNoteHandler}
        onDrop={onDrop}
        isOver={isOver}
        canDrop={canDrop}
        connectDropTarget={identity}
    />);

});

test('Should render corectly', () => {

    expect(wrapper).toMatchSnapshot();
})