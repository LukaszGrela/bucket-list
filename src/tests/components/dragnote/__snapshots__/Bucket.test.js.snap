// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render corectly 1`] = `
<div
  className="bucket"
>
  <div
    className="bucket-name"
  />
  <div
    className="notes-container"
  >
    <div
      className="add-note"
    >
      <AddBucketButton
        handleClick={[Function]}
      />
    </div>
  </div>
  <div
    className="bucket-toolbar"
  >
    Add Note
  </div>
</div>
`;
